### PR TITLE
[WIP] Basic SP and FP Algs

### PR DESCRIPTION
**Pull request by ericthewry**
This is a work in progress. Here's a first pass at the important algs that we have. Theyre just modified versions of BFS and Dijkstra. So should be mostly right.

Feel free to leave comments, but I still need to add more info.

Waiting on #4 to merge.

### CODE REVIEW

**Comment by rwoll** [comment]
@ericthewry Let me know if you have time to Skype about some of the algs in this document sometime between 11:00 AM PDT - 9:00 PM PDT.

**Comment by rwoll** [comment]
@ericthewry Let me know when you've had a chance to make the corrections to the algs and their runtimes after based on our Skype call and I will review again. :+1:

**Comment by ericthewry** [comment]
@rwoll Feel free to take another look

File: paths.tex [hunk 1]
```diff
@@ -331,24 +332,30 @@ \section{ Shortest Path Algorithms}
 of an edge, and have \texttt{edge\_neighborhood} pull it out, giving $O(1)$, but
 requiring precomputation with cost $O(m\log m)$.
 Each algorithm iterates through every vertex, and then in the
-worst case, every edge will enter $R$ at some point. So we have $O(nm\log m)$.
+worst case, every edge will enter $R$ at some point, giving $O(nm \log m)$. Then
+we do this accross all time slices, so we have $O(tnm\log m)$.
```
Comment (rwoll) [inline paths.tex]: Thanks for adjusting this as well. I wish it was as fast as the original `O` times, but timestamps do add just enough complexity to add the `t` multiplier. Do you think it is worth mentioning that for sparse graphs and real-life datasets this `O` time is truly an upper-bound and unlikely to be observed or should that be in a separate document once we run the algs?

File: paths.tex [hunk 2]
```diff
@@ -401,36 +408,33 @@ \section{Fastest Path Algorithms}
 These algorithms rely on a slightly different (and more correct) definition of
 fastest temporal path, where the time taken by a path is the absolute value of
 the difference between the latest start time within the active interval of the
-first edge, and the maximum of the start times of all edges.
+first edge, and the maximum of the start times of all edges. In the following
+algorithm, the lower bound is fixed by the initial starting time, so all we care
+about is the upper bound on the interval, because the time a $t,u,v$-path takes
+is the upper bound of the interval taken by all of the paths less $t$.
```
Comment (rwoll) [inline paths.tex]: Good to add this clariiying piece :+1:.

**Pull request merged**